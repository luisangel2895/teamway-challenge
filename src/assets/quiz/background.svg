<svg width="374" height="801" viewBox="0 0 374 801" fill="none" xmlns="http://www.w3.org/2000/svg">
<mask id="mask0_178_8748" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="0" y="-11" width="375" height="812">
<rect y="-11" width="375" height="812" fill="#6A5AE0"/>
</mask>
<g mask="url(#mask0_178_8748)">
<path opacity="0.1" fill-rule="evenodd" clip-rule="evenodd" d="M28.4375 231.25C60.3665 231.25 86.25 205.366 86.25 173.438C86.25 141.509 60.3665 115.625 28.4375 115.625C-3.49146 115.625 -29.375 141.509 -29.375 173.438C-29.375 205.366 -3.49146 231.25 28.4375 231.25Z" fill="#C4D0FB"/>
<path opacity="0.2" fill-rule="evenodd" clip-rule="evenodd" d="M30 275C85.2285 275 130 230.228 130 175C130 119.772 85.2285 75 30 75C-25.2285 75 -70 119.772 -70 175C-70 230.228 -25.2285 275 30 275Z" stroke="#C4D0FB"/>
<path opacity="0.1" fill-rule="evenodd" clip-rule="evenodd" d="M330.438 58.25C362.366 58.25 388.25 32.3665 388.25 0.4375C388.25 -31.4915 362.366 -57.375 330.438 -57.375C298.509 -57.375 272.625 -31.4915 272.625 0.4375C272.625 32.3665 298.509 58.25 330.438 58.25Z" fill="#C4D0FB"/>
<path opacity="0.2" fill-rule="evenodd" clip-rule="evenodd" d="M332 102C387.228 102 432 57.2285 432 2C432 -53.2285 387.228 -98 332 -98C276.772 -98 232 -53.2285 232 2C232 57.2285 276.772 102 332 102Z" stroke="#C4D0FB"/>
</g>
</svg>
