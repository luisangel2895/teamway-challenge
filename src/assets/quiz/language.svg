<svg width="64" height="64" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_296_6853)">
<rect width="64" height="64" rx="20" fill="#FFC2CD"/>
<rect x="8" y="16" width="48" height="66" rx="8" fill="white"/>
<path d="M19.3333 24L17 24C16.4477 24 16 24.4477 16 25L16 55C16 55.5523 16.4477 56 17 56L19.3333 56" stroke="#FFC2CC" stroke-width="3" stroke-linecap="round"/>
<rect x="23" y="30.6667" width="6" height="6" rx="3" fill="#FFD6DD"/>
<rect x="23" y="42.6667" width="6" height="6" rx="3" fill="#FFD6DD"/>
<rect x="35" y="30.6667" width="6" height="6" rx="3" fill="#FFD6DD"/>
<rect x="35" y="42.6667" width="6" height="6" rx="3" fill="#FFD6DD"/>
<path d="M44.6667 24L47 24C47.5523 24 48 24.4477 48 25L48 55C48 55.5523 47.5523 56 47 56L44.6667 56" stroke="#FFC2CC" stroke-width="3" stroke-linecap="round"/>
</g>
<defs>
<clipPath id="clip0_296_6853">
<rect width="64" height="64" rx="20" fill="white"/>
</clipPath>
</defs>
</svg>
