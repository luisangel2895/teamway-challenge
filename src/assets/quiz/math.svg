<svg width="64" height="64" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_296_6677)">
<rect width="64" height="64" rx="20" fill="#C4D0FB"/>
<rect x="8" y="16" width="48" height="66" rx="8" fill="white"/>
<path d="M34.3333 37.6667H35.5C36.6666 37.6667 36.6666 38.8334 37.852 41.7816C39 44.6668 39 45.8334 40.1667 45.8334H41.3333" stroke="#6A5AE0" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M33.1667 45.8334C34.9167 45.8334 36.6667 43.5001 37.8334 41.7501C39 40.0001 40.75 37.6667 42.5 37.6667" stroke="#6A5AE0" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M21.5 48.1667C21.5 49.9167 22.0833 50.5 23.8333 50.5C25.5833 50.5 26.1667 45.8333 27.3333 40C28.5 34.1667 29.0833 29.5 30.8333 29.5C32.5833 29.5 33.1667 30.0833 33.1667 31.8333" stroke="#6A5AE0" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M23.8333 40H30.8333" stroke="#6A5AE0" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
</g>
<defs>
<clipPath id="clip0_296_6677">
<rect width="64" height="64" rx="20" fill="white"/>
</clipPath>
</defs>
</svg>
