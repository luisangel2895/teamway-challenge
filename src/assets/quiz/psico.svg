<svg width="64" height="64" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_296_7602)">
<rect width="64" height="64" rx="20" fill="#C4D0FB"/>
<path d="M16 24C16 19.5817 19.5817 16 24 16H70V74C70 78.4183 66.4183 82 62 82H24C19.5817 82 16 78.4183 16 74V24Z" fill="white"/>
<rect x="25" y="40" width="6" height="32" rx="3" fill="#6A5AE0"/>
<rect x="37" y="36" width="6" height="47" rx="3" fill="#C4D0FB"/>
<rect x="49" y="32" width="6" height="41" rx="3" fill="#6A5AE0"/>
</g>
<defs>
<clipPath id="clip0_296_7602">
<rect width="64" height="64" rx="20" fill="white"/>
</clipPath>
</defs>
</svg>
