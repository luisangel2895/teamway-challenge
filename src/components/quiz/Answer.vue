<template lang="pug">
.answer
  .answer__number
    .answer__number-circle {{question.id}}
  .answer__response
    .answer__response-question {{question.question}}
    .answer__response-alternative {{answer.id}}. {{answer.alternative}}
</template>

<script lang="ts">
import { Options, Vue } from 'vue-class-component';

@Options({
  props: {
    answer: {
      type: Object,
    },
    question: {
      type: Object,
    },
  },
})
export default class Answer extends Vue {}
</script>

<style lang="scss" scoped>
.answer {
  width: 100%;
  display: flex;
  align-items: center;
  margin: 10px 0px;

  &__number {
    &-circle {
      width: 32px;
      height: 32px;
      margin: 0px 20px;
      display: flex;
      justify-content: center;
      align-items: center;
      background-color: $white_color;
      border-radius: 100%;
      border: 1px solid $grey_color_2;
      font-size: 1.6rem;
      font-weight: 600;
      color: $primary_color;
    }
  }

  &__response {
    margin: 0px 20px 0px 0px;

    &-question {
      font-size: 1.4rem;
      font-weight: 600;
      color: $black_color;
    }

    &-alternative {
      font-size: 1.4rem;
      color: $grey_color_2;
    }
  }
}
</style>
