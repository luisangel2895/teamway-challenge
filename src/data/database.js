const allQuizzes = [
  {
    id: 'test-personality',
    name: 'Personality Test',
    type: 'Psichology',
    number: 1,
    image:
      'https://image.shutterstock.com/image-illustration/abstract-shape-psychological-test-board-260nw-182162528.jpg',
  },
  {
    id: 'test-math',
    name: 'Math Test',
    type: 'Math',
    number: 2,
    image:
      'https://www.creativefabrica.com/wp-content/uploads/2019/08/Math-test-with-a-B-580x386.jpg',
  },
];

const quizzes = [
  {
    id: 'test-personality',
    questions: [
      {
        id: '1',
        question:
          "You're really busy at work and a colleague is telling you their life story and personal woes. You:",
        alternatives: [
          {
            id: 'a',
            alternative: "Don't dare to interrupt them",
          },
          {
            id: 'b',
            alternative:
              "Think it's more important to give them some of your time; work can wait",
          },
          {
            id: 'c',
            alternative: 'Listen, but with only with half an ear',
          },
          {
            id: 'd',
            alternative:
              'Interrupt and explain that you are really busy at the moment',
          },
        ],
      },
      {
        id: '2',
        question:
          "You've been sitting in the doctor's waiting room for more than 25 minutes. You:",
        alternatives: [
          {
            id: 'a',
            alternative: 'Look at your watch every two minutes',
          },
          {
            id: 'b',
            alternative: 'Bubble with inner anger, but keep quiet',
          },
          {
            id: 'c',
            alternative:
              'Explain to other equally impatient people in the room that the doctor is always running late',
          },
          {
            id: 'd',
            alternative:
              'Complain in a loud voice, while tapping your foot impatiently',
          },
        ],
      },
      {
        id: '3',
        question:
          "You're having an animated discussion with a colleague regarding a project that you're in charge of. You:",
        alternatives: [
          {
            id: 'a',
            alternative: "Don't dare contradict them",
          },
          {
            id: 'b',
            alternative: 'Think that they are obviously right',
          },
          {
            id: 'c',
            alternative: 'Defend your own point of view, tooth and nail',
          },
          {
            id: 'd',
            alternative: 'Continuously interrupt your colleague',
          },
        ],
      },
      {
        id: '4',
        question: 'You are taking part in a guided tour of a museum. You:',
        alternatives: [
          {
            id: 'a',
            alternative:
              "Are a bit too far towards the back so don't really hear what the guide is saying",
          },
          {
            id: 'b',
            alternative: 'Follow the group without question',
          },
          {
            id: 'c',
            alternative: 'Make sure that everyone is able to hear properly',
          },
          {
            id: 'd',
            alternative:
              'Are right up the front, adding your own comments in a loud voice',
          },
        ],
      },
      {
        id: '5',
        question:
          'During dinner parties at your home, you have a hard time with people who:',
        alternatives: [
          {
            id: 'a',
            alternative: 'Ask you to tell a story in front of everyone else',
          },
          {
            id: 'b',
            alternative: 'Talk privately between themselves',
          },
          {
            id: 'c',
            alternative: 'Hang around you all evening',
          },
          {
            id: 'd',
            alternative: 'Always drag the conversation back to themselves',
          },
        ],
      },
      {
        id: '6',
        question:
          'You crack a joke at work, but nobody seems to have noticed. You:',
        alternatives: [
          {
            id: 'a',
            alternative: "Think it's for the best — it was a lame joke anyway",
          },
          {
            id: 'b',
            alternative: 'Wait to share it with your friends after work',
          },
          {
            id: 'c',
            alternative: 'Try again a bit later with one of your colleagues',
          },
          {
            id: 'd',
            alternative: 'Keep telling it until they pay attention',
          },
        ],
      },
      {
        id: '7',
        question: 'This morning, your agenda seems to be free. You:',
        alternatives: [
          {
            id: 'a',
            alternative:
              'Know that somebody will find a reason to come and bother you',
          },
          {
            id: 'b',
            alternative:
              'Heave a sigh of relief and look forward to a day without stress',
          },
          {
            id: 'c',
            alternative:
              "Question your colleagues about a project that's been worrying you",
          },
          {
            id: 'd',
            alternative:
              'Pick up the phone and start filling up your agenda with meetings',
          },
        ],
      },
      {
        id: '8',
        question:
          'During dinner, the discussion moves to a subject about which you know nothing at all. You:',
        alternatives: [
          {
            id: 'a',
            alternative:
              "Don't dare show that you don't know anything about the subject",
          },
          {
            id: 'b',
            alternative: 'Barely follow the discussion',
          },
          {
            id: 'c',
            alternative: 'Ask lots of questions to learn more about it',
          },
          {
            id: 'd',
            alternative: 'Change the subject of discussion',
          },
        ],
      },
      {
        id: '9',
        question:
          "You're out with a group of friends and there's a person who's quite shy and doesn't talk much. You:",
        alternatives: [
          {
            id: 'a',
            alternative:
              "Notice that they're alone, but don't go over to talk with them",
          },
          {
            id: 'b',
            alternative: 'Go and have a chat with them',
          },
          {
            id: 'c',
            alternative: 'Shoot some friendly smiles in their direction',
          },
          {
            id: 'd',
            alternative: 'Hardly notice them at all',
          },
        ],
      },
      {
        id: '10',
        question:
          'At work, somebody asks for your help for the hundredth time. You:',
        alternatives: [
          {
            id: 'a',
            alternative: 'Give them a hand, as usual',
          },
          {
            id: 'b',
            alternative: "Accept — you're known for being helpful",
          },
          {
            id: 'c',
            alternative: 'Ask them, please, to find somebody else for a change',
          },
          {
            id: 'd',
            alternative: "Loudly make it known that you're annoyed",
          },
        ],
      },
      {
        id: '11',
        question:
          "You've been see a movie with your family and the reviews are mixed. You:",
        alternatives: [
          {
            id: 'a',
            alternative: "Don't share your point of view with anyone",
          },
          {
            id: 'b',
            alternative:
              "Didn't like the film, but keep your views to yourself when asked",
          },
          {
            id: 'c',
            alternative: 'State your point of view with enthusiasm',
          },
          {
            id: 'd',
            alternative: 'Try to bring the others round to your point of view',
          },
        ],
      },
      {
        id: '12',
        question: 'A friend arrives late for your meeting. You:',
        alternatives: [
          {
            id: 'a',
            alternative:
              "Say, 'It's not a problem,' even if that's not what you really think",
          },
          {
            id: 'b',
            alternative:
              'Give them a filthy look and sulk for the rest of the evening',
          },
          {
            id: 'c',
            alternative:
              "Tell them, 'You're too much! Have you seen the time?'",
          },
          {
            id: 'd',
            alternative: 'Make a scene in front of everyone',
          },
        ],
      },
      {
        id: '13',
        question: "You can't find your car keys. You:",
        alternatives: [
          {
            id: 'a',
            alternative:
              "Don't want anyone to find out, so you take the bus instead",
          },
          {
            id: 'b',
            alternative:
              'Panic and search madly without asking anyone for help',
          },
          {
            id: 'c',
            alternative:
              "Grumble without telling your family why you're in a bad mood",
          },
          {
            id: 'd',
            alternative: 'Accuse those around you for misplacing them',
          },
        ],
      },
      {
        id: '14',
        question: "It's time for your annual appraisal with your boss. You:",
        alternatives: [
          {
            id: 'a',
            alternative:
              'Go with great hesitation as these sessions are torture for you',
          },
          {
            id: 'b',
            alternative:
              'Look forward to hearing what your boss thinks about you and expects from you',
          },
          {
            id: 'c',
            alternative:
              "Rehearse ad nauseam the arguments and ideas that you've prepared for the meeting",
          },
          {
            id: 'd',
            alternative:
              'Go along unprepared as you are confident and like improvising',
          },
        ],
      },
    ],
  },
  {
    id: 'test-math',
    questions: [
      {
        id: '1',
        question: 'what is 1 + 1',
        alternatives: [
          {
            id: 'a',
            alternative: '1',
          },
          {
            id: 'b',
            alternative: '2',
          },
          {
            id: 'c',
            alternative: '3',
          },
          {
            id: 'd',
            alternative: '4',
          },
        ],
      },
      {
        id: '2',
        question: 'what is 2 x 2',
        alternatives: [
          {
            id: 'a',
            alternative: '1',
          },
          {
            id: 'b',
            alternative: '2',
          },
          {
            id: 'c',
            alternative: '3',
          },
          {
            id: 'd',
            alternative: '4',
          },
        ],
      },
      {
        id: '3',
        question: 'what is 5 / 2',
        alternatives: [
          {
            id: 'a',
            alternative: '1',
          },
          {
            id: 'b',
            alternative: '2',
          },
          {
            id: 'c',
            alternative: '2.5',
          },
          {
            id: 'd',
            alternative: '3.5',
          },
        ],
      },
      {
        id: '4',
        question: 'what is 345 - 45',
        alternatives: [
          {
            id: 'a',
            alternative: '255',
          },
          {
            id: 'b',
            alternative: '300',
          },
          {
            id: 'c',
            alternative: '355',
          },
          {
            id: 'd',
            alternative: '200',
          },
        ],
      },
      {
        id: '5',
        question: 'what is 55 / 5',
        alternatives: [
          {
            id: 'a',
            alternative: '11',
          },
          {
            id: 'b',
            alternative: '10',
          },
          {
            id: 'c',
            alternative: '2',
          },
          {
            id: 'd',
            alternative: '6',
          },
        ],
      },
    ],
  },
];

const responses = [
  {
    quiz: 'test-personality',
    results: [
      {
        id: 'introvert',
        title: 'You are more of an introvert',
        text: `You feel that living alone is to live happily, and you prefer hiding in a crowd rather than standing out in one.

        You are perpetually tormented by the idea of doing things wrong, not understanding or not being alert enough or intelligent 
        enough to do what others expect of you. You lack in self-confidence and you seem to believe that others are better than you.
         While in a conversation, for example, you would be more likely to go along with the other's points of view as you don't fully 
         respect your own opinions. Where there's a low level task to complete or a service to be allotted, it's you who volunteers. 
         When people want to get out of a task, they naturally come to you as they know that you never say 'no'.

        It's not surprising that you sometimes have the impression of being permanently exploited, but you don't really know how to 
        break this vicious circle. On the rare occasions that you've tried to do so, this uncharacteristic defiance has caused uproar 
        and you've ended up backing down. You are afraid of contact with others as you imagine that they are constantly judging you and 
        that their probing looks will obviously find your faults. So, you try to remain the most transparent and discreet possible. How 
        far will you take this logic of self-denigration? Try and stop projecting onto others the bad image you have about yourself. 

        Have a good look around you and you'll see that they too have faults, weaknesses and shortcomings, so stop finding excuses for everything. 
        Maybe you were brought up in an atmosphere of 'You'll never make the grade' as a child — a poison that you need progressively to 
        get out of your system. Learn to look after your own interests — everyone else does, so why not you? You too have desires, dreams and 
        opinions — express them. If you think that by saying no or thinking differently from others that you will no longer be loved, it just isn't so. 
        Others will learn to respect you because you respect yourself.`,
      },
      {
        id: 'extrovert',
        title: 'You are more of a public introvert and private extrovert',
        text: `Within your circle of family and friends, you are completely at ease and it's often you who takes the lead to organise outings, dinners, 
        vacations, etc. However, as soon as you are in a public or professional setting you become rather inhibited. You don't feel in tune with what 
        you would qualify as agitation, rivalry or vainglory and this can sometimes appear to others as a kind of hidden criticism or a lack of ambition. 
        In fact, your pragmatic temperament helps you successfully manage problems as and when they arise. You aren't the kind to calculate your every 
        move and plan every inch of your life, and you find the sphere of family and friends to be much more gratifying to the professional sphere. You 
        are more expressive in emotional, sensorial and human terms and at home your organisation skills and capacity to oil the family wheels work well 
        and many of your family and friends appreciate your qualities and skills.

        At first glance, people find it hard to understand how you could be so comfortable organising your private life and then seem to lose the better 
        part of your self-confidence when you're in public. Maybe it's a question of rhythm? If you feel comfortable in your domestic rhythm it's perhaps 
        because it works more on a short-term basis (day, week or more rarely a month). On the contrary, the rhythm of professional life seems less concrete 
        and more distant as the professional agenda works more to quarterly or annual plans, that you have no control over. Is this the source of your lessened 
        motivation? You can't control all the cards so you have to adapt and, in fact, that doesn't interest you at all? If this difference between work and 
        home life doesn't bother you, then carry on. However, if you feel frustrated by this imbalance, it could be useful to look into the deeper reasons 
        (either on your own or with someone else) that keep you less focused on professional ambition. For instance, such reasons could be: a lack of thanks 
        and recognition, a fear of not being up to the mark… It's up to you to find out.`,
      },
    ],
  },
  {
    quiz: 'test-math',
    results: [
      {
        id: 'good',
        title: "You're a good in math",
        text: "You're a good in math, congratulations, few people have talent in math and you are a one, in the new update of the app we include quizzes of advanced math especialy for talent people like you.",
      },
      {
        id: 'bad',
        title: 'You need a more practice',
        text: 'You need a more practice, but you can improve in math,you just need to make tests every day in our platform, in in just 1 month you can become in a master in math.',
      },
    ],
  },
];

export { allQuizzes, quizzes, responses };
