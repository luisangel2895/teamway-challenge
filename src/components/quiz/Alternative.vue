<template lang="pug">
.alternative(:class="{selected : selected}" @click="selectAlternative")
    | {{alternative.id}}. {{alternative.alternative}}
</template>

<script lang="ts">
import { Options, Vue } from 'vue-class-component';
import { mapMutations, mapState } from 'vuex';
// Types
import { Alternative as AlternativeType } from '@/types/quiz.d';

@Options({
  computed: mapState(['alternativeSelected']),
  props: {
    alternative: {
      type: Object,
    },
  },
  watch: {
    alternativeSelected(value: AlternativeType) {
      if (this.alternative.id === value.id) {
        this.selected = true;
      } else {
        this.selected = false;
      }
    },
  },
  methods: mapMutations(['UPDATE_ALTERNATIVE_SELECTED']),
})
export default class Alternative extends Vue {
  alternative!: AlternativeType;

  alternativeSelected!: AlternativeType;

  selected = false;

  UPDATE_ALTERNATIVE_SELECTED!: (alternative: AlternativeType) => void;

  selectAlternative(): void {
    this.UPDATE_ALTERNATIVE_SELECTED(this.alternative);
  }
}
</script>

<style lang="scss" scoped>
.alternative {
  width: 100%;
  margin-top: 20px;
  padding: 15px;
  background-color: $white_color;
  border-radius: 10px;
  border: 2px solid $grey_color_5;
  color: $black_color;
  font-size: 1.6rem;
}
.selected {
  font-weight: 600;
  background-color: $grey_color_5;
}
</style>
