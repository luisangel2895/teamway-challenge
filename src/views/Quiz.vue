<template lang="pug">
.quiz-page
  .container-progress
    .container-progress__profile
      img(src="@/assets/home/profile.jpg")
    .container-progress__bar
      #bar.container-progress__bar-filled
    .container-progress__percent
      .container-progress__percent-box {{percentQuiz}} %
  .container-questions
    Question(:question="currentQuiz.questions[id-1]" @next="handleQuestion"
      :total="currentQuiz.questions.length")
</template>

<script lang="ts">
import { Options, Vue } from 'vue-class-component';
import { mapState } from 'vuex';
// Components
import Question from '@/components/quiz/Question.vue';
// Types
import { Quiz as QuizType } from '@/types/quiz.d';

@Options({
  computed: {
    ...mapState(['currentQuiz', 'percentQuiz']),
  },
  components: {
    Question,
  },
})
export default class Quiz extends Vue {
  id = 1;

  currentQuiz!: QuizType;

  percentQuiz!: number;

  handleQuestion(value: string): void {
    if (this.id >= this.currentQuiz.questions.length) {
      this.$router.push({ name: 'Result' });
    } else {
      this.id = Number(value) + 1;
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      document.getElementById('bar').style.width = `${this.percentQuiz}%`;
    }
  }
}
</script>

<style lang="scss" scoped>
.quiz-page {
  width: 100vw;
  height: 100vh;
  position: absolute;
  z-index: 20;
  display: flex;
  flex-direction: column;
  background-color: $primary_color;
  font-family: $text_font;
}
.container-progress {
  height: 10%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  &__profile {
    width: 25%;
    text-align: center;
    img {
      width: 50px;
      border-radius: 100%;
    }
  }

  &__bar {
    width: 50%;
    height: 5px;
    background-color: rgba(255, 255, 255, 0.3);
    text-align: left;

    &-filled {
      width: 0%;
      height: 5px;
      background-color: rgb(255, 255, 255);
    }
  }

  &__percent {
    width: 25%;
    text-align: center;
    &-box {
      width: 50px;
      margin: auto;
      padding: 10px 0px 10px 0px;
      background-color: #ff9b57;
      font-size: 1.2rem;
      font-weight: 600;
      color: white;
      border-radius: 10px;
    }
  }
}
.container-questions {
  height: 90%;
  padding: 0px 10px 10px 10px;
}
</style>
