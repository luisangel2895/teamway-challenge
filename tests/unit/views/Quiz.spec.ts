import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
// Components
import Quiz from '@/views/Quiz.vue';
// Types
import { Quiz as QuizType } from '@/types/quiz.d';
// Fake Data
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { quizzes } from '@/data/database';

interface State {
  currentQuiz: QuizType;
  percentQuiz: number;
}
const store = createStore({
  state(): State {
    return {
      currentQuiz: quizzes[0] as QuizType,
      percentQuiz: 0,
    };
  },
});

const wrapper = shallowMount(Quiz, {
  global: {
    plugins: [store],
  },
});

describe('Test in component Quiz.vue', (): void => {
  it('Match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Question component mounted into this component', (): void => {
    expect(wrapper.find('question-stub').exists()).toBeTruthy();
  });
  it('Recognize correctly the number of questions of the test', (): void => {
    expect(wrapper.find('question-stub').attributes('total')).toBe('14');
  });
});
