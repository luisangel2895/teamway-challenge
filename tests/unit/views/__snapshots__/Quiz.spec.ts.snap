// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component Quiz.vue Match with snapshot 1`] = `
<div class="quiz-page">
  <div class="container-progress">
    <div class="container-progress__profile"><img></div>
    <div class="container-progress__bar">
      <div class="container-progress__bar-filled" id="bar"></div>
    </div>
    <div class="container-progress__percent">
      <div class="container-progress__percent-box">0 %</div>
    </div>
  </div>
  <div class="container-questions">
    <question-stub question="[object Object]" total="14"></question-stub>
  </div>
</div>
`;
